applications/leaderboard/Leaderboard/Program.cs

```diff
@@ -0,0 +1,525 @@
+                var cmd = connection.CreateDdlCommand(createStatement);
+                try
+                {
+                    await cmd.ExecuteNonQueryAsync();
+                }
+                catch (SpannerException e) when (e.ErrorCode == ErrorCode.AlreadyExists)
+                {
+                    // OK.
+                }
+            }
+            // Update connection string with Database ID for table creation.
+            connectionString = connectionString + $"/databases/{databaseId}";
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                // Define create table statement for table #1.
+                string createTableStatement =
+               @"CREATE TABLE Players(
+                    PlayerId INT64 NOT NULL,
+                    PlayerName STRING(2048) NOT NULL
+                ) PRIMARY KEY(PlayerId)";
+                // Make the request.
+                var cmd = connection.CreateDdlCommand(createTableStatement);
+                await cmd.ExecuteNonQueryAsync();
+                // Define create table statement for table #2.
+                createTableStatement =
+                @"CREATE TABLE Scores(
+                    PlayerId INT64 NOT NULL,
+                    Score INT64 NOT NULL,
+                    Timestamp TIMESTAMP NOT NULL OPTIONS(allow_commit_timestamp=true)
+                ) PRIMARY KEY(PlayerId, Timestamp), INTERLEAVE IN PARENT Players ON DELETE NO ACTION";
+                // Make the request.
+                cmd = connection.CreateDdlCommand(createTableStatement);
+                await cmd.ExecuteNonQueryAsync();
+            }
+        }
+
+        public static object InsertPlayers(string projectId,
+            string instanceId, string databaseId)
+        {
+            var responseTask =
+                InsertPlayersAsync(projectId, instanceId, databaseId);
+            Console.WriteLine("Waiting for operation to complete...");
+            responseTask.Wait();
+            Console.WriteLine($"Operation status: {responseTask.Status}");
+            Console.WriteLine($"Inserted records into sample database "
+                + $"{databaseId} on instance {instanceId}");
+            return ExitCode.Success;
+        }
+
+        public static async Task InsertPlayersAsync(string projectId,
+            string instanceId, string databaseId)
+        {
+            string connectionString =
+                $"Data Source=projects/{projectId}/instances/{instanceId}"
+                + $"/databases/{databaseId}";
+
+            // Get current max PlayerId.
+            Int64 maxPlayerId = 0;
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                // Execute a SQL statement to get current MAX() of PlayerId.
+                var cmd = connection.CreateSelectCommand(
+                    @"SELECT MAX(PlayerId) as PlayerId FROM Players");
```

Comment:
Is this really the recommended way in spanner?  In other databases, I know this pattern creates a bottleneck on the playerId.  In SqlServer, for example, I would expect this value to be a GUID.